src/inventory_ui.cpp

```diff
@@ -449,30 +691,44 @@ inventory_entry *inventory_selector::find_entry_by_invlet( long invlet ) const
     return nullptr;
 }
 
-void inventory_selector::prepare_columns( bool multiselect )
+void inventory_selector::rearrange_columns()
 {
-    for( auto &column : columns ) {
-        column->set_multiselect( multiselect );
+    if( !own_gear_column.empty() && is_overflown() ) {
+        own_gear_column.move_entries_to( own_inv_column );
     }
 
-    if( custom_column != nullptr ) {
-        if( columns.empty() || column_can_fit( *custom_column ) ) {
-            // Make the column second if possible
-            const auto position = ( !columns.empty() ) ? std::next( columns.begin() ) : columns.begin();
+    if( !map_column.empty() && is_overflown() ) {
+        map_column.move_entries_to( own_inv_column );
+    }
+}
 
-            custom_column->set_multiselect( multiselect );
-            insert_column( position, custom_column );
-        } else {
-            columns.front()->add_entries( *custom_column );
-            custom_column.release();
-        }
+void inventory_selector::prepare_layout()
+{
+    if( layout_is_valid ) {
+        return;
+    }
+    // This block adds categories and should go before any width evaluations
+    for( auto &column : columns ) {
+        column->prepare_paging();
+    }
+    // Handle screen overflow
+    rearrange_columns();
+    // If we have a single column and it occupies more than a half of
+    // the available with -> expand it
+    auto visible_columns = get_visible_columns();
+    if( visible_columns.size() == 1 && are_columns_centered() ) {
+        visible_columns.front()->set_width( getmaxx( w_inv ) - 2 * screen_border_gap );
     }
 
+    long custom_invlet = '0';
     for( auto &column : columns ) {
-        column->prepare_paging( getmaxy( w_inv ) - 5 );
+        column->prepare_paging();
```

Comment:
And again here. Singular vs plural variable names are a good source of (unnecessary) confusion.